scripts/check_ecosystem.py

```diff
@@ -46,14 +52,22 @@ async def clone(self: Self) -> "AsyncIterator[Path]":
                 "--depth",
                 "1",
                 "--no-tags",
-                "--branch",
-                self.ref,
-                f"https://github.com/{self.org}/{self.repo}",
-                tmpdir,
+            ]
+            if self.ref:
+                git_command.extend(["--branch", self.ref])
+            git_command.extend(
```

Comment:
Nit: Add a newline to make it clearer that this statement does not belong to the if body?
```suggestion
                git_command.extend(["--branch", self.ref])

          	git_command.extend(
```